In [15]:
def read_csv(path):

    with open(path, 'r+') as file:
        results = []
        for line in file:
            line = line.rstrip('\n') # remove `\n` at the end of line
            items = line.split(',')
            results.append(tuple(items))
        # after for-loop
        return results

In [16]:
df = read_csv('/content/airlines.csv')
df

[('Airport.Code', 'Airport.Name', 'Time.Year', 'Statistics.Flights.Cancelled'),
 ('ATL',
  '"Atlanta',
  ' GA: Hartsfield-Jackson Atlanta International"',
  '2003',
  '216'),
 ('BOS', '"Boston', ' MA: Logan International"', '2003', '138'),
 ('BWI',
  '"Baltimore',
  ' MD: Baltimore/Washington International Thurgood Marshall"',
  '2003',
  '29'),
 ('CLT', '"Charlotte', ' NC: Charlotte Douglas International"', '2003', '73'),
 ('DCA',
  '"Washington',
  ' DC: Ronald Reagan Washington National"',
  '2003',
  '74'),
 ('DEN', '"Denver', ' CO: Denver International"', '2003', '34'),
 ('DFW',
  '"Dallas/Fort Worth',
  ' TX: Dallas/Fort Worth International"',
  '2003',
  '394'),
 ('DTW', '"Detroit', ' MI: Detroit Metro Wayne County"', '2003', '123'),
 ('EWR', '"Newark', ' NJ: Newark Liberty International"', '2003', '102'),
 ('FLL',
  '"Fort Lauderdale',
  ' FL: Fort Lauderdale-Hollywood International"',
  '2003',
  '13'),
 ('IAD',
  '"Washington',
  ' DC: Washington Dulles International"',
  '20

In [28]:
df[0]

('Airport.Code', 'Airport.Name', 'Time.Year', 'Statistics.Flights.Cancelled')

In [32]:
df[1][1:3]

('"Atlanta', ' GA: Hartsfield-Jackson Atlanta International"')

In [62]:
airport = [] 
for i, row in enumerate(df):
  airport.append(df[i][1:3])
  # print(df[i][1:3])

In [40]:
airport

[('Airport.Name', 'Time.Year'),
 ('"Atlanta', ' GA: Hartsfield-Jackson Atlanta International"'),
 ('"Boston', ' MA: Logan International"'),
 ('"Baltimore', ' MD: Baltimore/Washington International Thurgood Marshall"'),
 ('"Charlotte', ' NC: Charlotte Douglas International"'),
 ('"Washington', ' DC: Ronald Reagan Washington National"'),
 ('"Denver', ' CO: Denver International"'),
 ('"Dallas/Fort Worth', ' TX: Dallas/Fort Worth International"'),
 ('"Detroit', ' MI: Detroit Metro Wayne County"'),
 ('"Newark', ' NJ: Newark Liberty International"'),
 ('"Fort Lauderdale', ' FL: Fort Lauderdale-Hollywood International"'),
 ('"Washington', ' DC: Washington Dulles International"'),
 ('"Houston', ' TX: George Bush Intercontinental/Houston"'),
 ('"New York', ' NY: John F. Kennedy International"'),
 ('"Las Vegas', ' NV: McCarran International"'),
 ('"Los Angeles', ' CA: Los Angeles International"'),
 ('"New York', ' NY: LaGuardia"'),
 ('"Orlando', ' FL: Orlando International"'),
 ('"Chicago', ' IL

In [43]:
c = {}
for x in airport:
  c[x] = c.get(x, 0) + 1
c

{('"Atlanta', ' GA: Hartsfield-Jackson Atlanta International"'): 154,
 ('"Baltimore',
  ' MD: Baltimore/Washington International Thurgood Marshall"'): 150,
 ('"Boston', ' MA: Logan International"'): 150,
 ('"Charlotte', ' NC: Charlotte Douglas International"'): 150,
 ('"Chicago', ' IL: Chicago Midway International"'): 150,
 ('"Chicago', ' IL: Chicago O\'Hare International"'): 150,
 ('"Dallas/Fort Worth', ' TX: Dallas/Fort Worth International"'): 150,
 ('"Denver', ' CO: Denver International"'): 150,
 ('"Detroit', ' MI: Detroit Metro Wayne County"'): 150,
 ('"Fort Lauderdale', ' FL: Fort Lauderdale-Hollywood International"'): 150,
 ('"Houston', ' TX: George Bush Intercontinental/Houston"'): 150,
 ('"Las Vegas', ' NV: McCarran International"'): 150,
 ('"Los Angeles', ' CA: Los Angeles International"'): 150,
 ('"Miami', ' FL: Miami International"'): 150,
 ('"Minneapolis', ' MN: Minneapolis-St Paul International"'): 150,
 ('"New York', ' NY: John F. Kennedy International"'): 150,
 ('"New Yo

In [58]:
dict_copy = {key[0] + key[1]: value for key, value in c.items()}

In [59]:
dict_copy

{'"Atlanta GA: Hartsfield-Jackson Atlanta International"': 154,
 '"Baltimore MD: Baltimore/Washington International Thurgood Marshall"': 150,
 '"Boston MA: Logan International"': 150,
 '"Charlotte NC: Charlotte Douglas International"': 150,
 '"Chicago IL: Chicago Midway International"': 150,
 '"Chicago IL: Chicago O\'Hare International"': 150,
 '"Dallas/Fort Worth TX: Dallas/Fort Worth International"': 150,
 '"Denver CO: Denver International"': 150,
 '"Detroit MI: Detroit Metro Wayne County"': 150,
 '"Fort Lauderdale FL: Fort Lauderdale-Hollywood International"': 150,
 '"Houston TX: George Bush Intercontinental/Houston"': 150,
 '"Las Vegas NV: McCarran International"': 150,
 '"Los Angeles CA: Los Angeles International"': 150,
 '"Miami FL: Miami International"': 150,
 '"Minneapolis MN: Minneapolis-St Paul International"': 150,
 '"New York NY: John F. Kennedy International"': 150,
 '"New York NY: LaGuardia"': 150,
 '"Newark NJ: Newark Liberty International"': 150,
 '"Orlando FL: Orlando 

In [64]:
dict_copy.pop('Airport.NameTime.Year')

1

In [69]:
dict_copy2 = {key[1:-2]: value for key, value in dict_copy.items()}
dict_copy2

{'Atlanta GA: Hartsfield-Jackson Atlanta Internationa': 154,
 'Baltimore MD: Baltimore/Washington International Thurgood Marshal': 150,
 'Boston MA: Logan Internationa': 150,
 'Charlotte NC: Charlotte Douglas Internationa': 150,
 'Chicago IL: Chicago Midway Internationa': 150,
 "Chicago IL: Chicago O'Hare Internationa": 150,
 'Dallas/Fort Worth TX: Dallas/Fort Worth Internationa': 150,
 'Denver CO: Denver Internationa': 150,
 'Detroit MI: Detroit Metro Wayne Count': 150,
 'Fort Lauderdale FL: Fort Lauderdale-Hollywood Internationa': 150,
 'Houston TX: George Bush Intercontinental/Housto': 150,
 'Las Vegas NV: McCarran Internationa': 150,
 'Los Angeles CA: Los Angeles Internationa': 150,
 'Miami FL: Miami Internationa': 150,
 'Minneapolis MN: Minneapolis-St Paul Internationa': 150,
 'New York NY: John F. Kennedy Internationa': 150,
 'New York NY: LaGuardi': 150,
 'Newark NJ: Newark Liberty Internationa': 150,
 'Orlando FL: Orlando Internationa': 150,
 'Philadelphia PA: Philadelphia Inte

In [71]:
import json 
with open("file.json", "w") as outputfile:
  json.dump(dict_copy2, outputfile)

#output 2 :
Atlanta GA: Hartsfield-Jackson Atlanta Internationa': 154 highest number of times and its count



In [72]:
maximum = max(dict_copy2, key=dict_copy2.get)
print(maximum, dict_copy2[maximum])

Atlanta GA: Hartsfield-Jackson Atlanta Internationa 154


#OUTPUT3-:
'Phoenix AZ: Phoenix Sky Harbor Internationa': 137,

In [73]:
minimum = min(dict_copy2, key=dict_copy2.get)
print(minimum, dict_copy2[minimum])

Phoenix AZ: Phoenix Sky Harbor Internationa 137
